src/modules/python/pbs/v1/_pmi_utils.py

```diff
@@ -113,8 +113,8 @@ def _get_hosts(job):
     not accept FQDNs.
     """
     hosts = str(job.exec_host2)
-    pbs_nodes = sorted(set([x.partition(':')[0].partition('.')[0]
-                            for x in hosts.split('+')]))
+    pbs_nodes = sorted({x.partition(':')[0].partition('.')[0]
```

Comment:
I think you mean `[]` instead `{}`

Reply:
I think this is correct. We want to "Form a list of unique short hostnames from a pbs job" according to the doc statement.
[nehap@neha20 ~]$ python3
Python 3.6.8 (default, Apr 25 2019, 21:02:35) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-36)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> hosts='host1.pbs.com:xyz+host2.pbs1.com1:abc+host3.pbs2.com2:def+host4.no:whatever'
>>> list1 = ({a.partition(':')[0].partition('.')[0] for a in hosts.split('+')})
>>> type(list1)
<class 'set'>
>>> sorted(list1)
['host1', 'host2', 'host3', 'host4']
>>> type(sorted(list1))
<class 'list'>
>>> 

[nehap@neha20 ~]$ python2
Python 2.7.5 (default, Jun 20 2019, 20:27:34) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-36)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> hosts='host1.pbs.com:xyz+host2.pbs1.com1:abc+host3.pbs2.com2:def+host4.no:whatever'
>>> type(([x.partition(':')[0].partition('.')[0] for x in hosts.split('+')]))
<type 'list'>
>>> set([x.partition(':')[0].partition('.')[0] for x in hosts.split('+')])
set(['host4', 'host3', 'host2', 'host1'])
>>> sorted(set([x.partition(':')[0].partition('.')[0] for x in hosts.split('+')]))
['host1', 'host2', 'host3', 'host4']
>>> type(sorted(set([x.partition(':')[0].partition('.')[0] for x in hosts.split('+')])))
<type 'list'>
>>>

Reply:
Oh yes, I forget using `{}` without key-value but with only value creates set in python. Sorry my mistake, please ignore this comment.